### ISSUE TITLE
`stax.BatchNorm` incorrect normalization

### ISSUE THREAD

**Issue by radugrosu** [google/jax#442]
jax.experimental.stax on current master, line 140 ( `BatchNorm` function):
```python
z = (x - mean) / (var + epsilon)**2
```
The square in the denominator should probably be a square root.

**Comment by mattjj** [comment]
Thanks so much for catching this!
